src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -628,16 +628,28 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                                 case CommandType.PUT:
                                     if (current < count)
                                     {
-                                        // delete the document to avoid exception if we put new document in a different collection.
-                                        // TODO: document this behavior
-                                        using (DocumentIdWorker.GetSliceFromId(context, cmd.Id, out Slice lowerId))
+                                        NonPersistentDocumentFlags nonPersistentDocumentFlags = NonPersistentDocumentFlags.None;
+
+                                        if (cmd.FromBackup)
+                                        {
+                                            // if the document came from backup it must have the same collection
+                                            // the only thing that we update is the change vector
+                                            // in this case, we can skip revision creation
+                                            nonPersistentDocumentFlags = NonPersistentDocumentFlags.SkipRevisionCreation;
+                                        }
+                                        else
                                         {
-                                            Database.DocumentsStorage.Delete(context, lowerId, cmd.Id, expectedChangeVector: null,
-                                                nonPersistentFlags: NonPersistentDocumentFlags.SkipRevisionCreation);
+                                            // delete the document to avoid exception if we put new document in a different collection.
```

Comment:
What about incremental backup restore?
1. In full we have a doc in X collection
2. In incremental it is in Y

Reply:
You are correct. Done.